src/rateEngine/LoadProfileScaler.ts

```diff
@@ -24,58 +34,69 @@ class LoadProfileScaler {
     return this.to(scaler);
   }
 
-  toAverageMonthlyBill(
-    amount: number,
-    rate: RateInterface
-  ): LoadProfile {
-    // Need this as a percent for better resolution within the algorithm
-    const scalerAsPercent = 100;
-    const fnParams = [scalerAsPercent, rate, this];
+  toAverageMonthlyBill(amount: number, rate: RateInterface, goalSeekParams: GoalSeekArgs = {}): LoadProfile {
+    const magnitude = Math.max(Math.floor(Math.log10(Math.abs(amount))), 0);
+
+    const magnitudeScaler = 10 ** magnitude;
+    const initialScalerGuess = magnitudeScaler;
+    const fnParams = [initialScalerGuess, rate, this, magnitude];
 
     try {
-      const finalScalerAsPercent = goalSeek({
+      const finalScaler = goalSeek({
         fn: this.scaledMonthlyCost,
         fnParams,
         percentTolerance: 0.1,
         maxIterations: 1000,
-        maxStep: 50,
+        maxStep: 0.5 * magnitudeScaler,
         goal: amount,
-        independentVariableIdx: 0
+        independentVariableIdx: 0,
+        ...goalSeekParams,
```

Comment:
This is also here to help "debug" scaling outside of the package. It might be useful at some point if a different use case needs a looser/tighter tolerance or can deal with a lot more iterations.